python/ray/air/tests/test_remote_storage_hdfs.py

```diff
@@ -0,0 +1,40 @@
+"""Test remote_storage in a ci environment with real hdfs setup."""
+import os
+
+import pytest
+
+from ray.air._internal.remote_storage import get_fs_and_path
+
+
+@pytest.fixture
+def set_hdfs_env_var_and_get_hostname():
+    """Set env vars required by pyarrow to talk to hdfs correctly.
+
+    Returns hostname that is needed for the hdfs uri."""
+
+    # the following file is written in `install-hdfs.sh`.
+    with open("/tmp/hdfs_env", "r") as f:
+        for line in f.readlines():
+            line = line.rstrip("\n")
+            tokens = line.split("=", maxsplit=1)
+            os.environ[tokens[0]] = tokens[1]
+    import sys
+
+    sys.path.insert(0, os.path.join(os.environ["HADOOP_HOME"], "bin"))
+    hostname = os.getenv("CONTAINER_ID")
+    yield hostname
+
+
+def test_get_fs_and_path_hdfs(set_hdfs_env_var_and_get_hostname):
+    hostname = set_hdfs_env_var_and_get_hostname
+    hdfs_uri = f"hdfs://{hostname}:8020/test/"
```

Comment:
Can we get the port from the fixture as well?

Reply:
done!